Add "atmos" COMPONENT subfolder to DMPDIR paths (dev_v16)

KateFriedman-NOAA opened the pull request:
**Description**

This PR adds the new `atmos` COMPONENT subfolder to relevant `DMPDIR` paths in workflow configs and scripts. Companion updates from the GSI, UFS_UTILS, and GLDAS will be coming as well.

`atmos` subfolder symlinks have been added to GDAs on WCOSS2, WCOSS1, Hera, and Orion to support versions of the system before this update. The WCOSS2 GDA will include an actual `atmos` subfolder and not a symlink as of the go-live cycle on WCOSS2. Versions of the GFS/global-workflow, which do not have the `atmos` subfolder in `DMPDIR` paths, will not work for GDA dates after WCOSS2 go-live but will continue working with pre-go-live dates in all GDAs.

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [x] Cycled test on WCOSS2 which reproduces WCOSS2 GFS prod.
  
Refs: #665, #802

KateFriedman-NOAA committed:
Add "atmos" COMPONENT subfolder to DMPDIR paths

Refs: #665, #802

aerorahul commented on parm/config/config.anal:
```diff
@@ -57,21 +57,22 @@ export OBERROR=$FIXgsi/prepobs_errtable.global
 
 
 # Use experimental dumps in EMC GFS v16 parallels
+export COMPONENT=${COMPONENT:-atmos}
```
This is atmospheric analysis job.  Is it ever going to get a value of `COMPONENT` from anywhere else and will that be anything other than `atmos`?
```suggestion
export COMPONENT=atmos
```

KateFriedman-NOAA replied:
Probably won't be anything other than `atmos`. Will accept your suggested change, thanks!

aerorahul commented on jobs/rocoto/prep.sh:
```diff
@@ -36,7 +36,7 @@ export COMOUT="$ROTDIR/$CDUMP.$PDY/$cyc/$COMPONENT"
 # If ROTDIR_DUMP=YES, copy dump files to rotdir 
 if [ $ROTDIR_DUMP = "YES" ]; then
-   $HOMEgfs/ush/getdump.sh $CDATE $CDUMP $DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc} $COMOUT
+   $HOMEgfs/ush/getdump.sh $CDATE $CDUMP $DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT} $COMOUT
```
Where does `prep.sh` get `$COMPONENT` from? `config.prep`?

KateFriedman-NOAA replied:
`prep.sh` has this nearer the top:
```
export COMPONENT=${COMPONENT:-atmos}
```
https://github.com/KateFriedman-NOAA/global-workflow/blob/feature/dev_v16_gdaatmos/jobs/rocoto/prep.sh#L31

aerorahul commented on parm/config/config.anal:
```diff
@@ -57,21 +57,22 @@ export OBERROR=$FIXgsi/prepobs_errtable.global
     export ABIBF="/dev/null"
     if [[ "$CDATE" -ge "2019022800" ]] ; then
-	export ABIBF="$DMPDIR/${CDUMP}x.${PDY}/${cyc}/${CDUMP}.t${cyc}z.gsrcsr.tm00.bufr_d"
+	export ABIBF="$DMPDIR/${CDUMP}x.${PDY}/${cyc}/${COMPONENT}/${CDUMP}.t${cyc}z.gsrcsr.tm00.bufr_d"
```
I don't see why these can't be wired to `atmos` since this is the atmospheric analysis and obs.

KateFriedman-NOAA replied:
I accepted the change to hardcode `COMPONENT=atmos` before this line in `config.anal` so this will be `atmos` now via the variable. Want me to hardcode to `atmos` here still?

aerorahul commented on ush/drive_makeprepbufr.sh:
```diff
@@ -55,7 +55,7 @@ GSUFFIX=${GSUFFIX:-$SUFFIX}
 ASUFFIX=${ASUFFIX:-$SUFFIX}
 
-COMIN_OBS=${COMIN_OBS:-"$DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}"}
+COMIN_OBS=${COMIN_OBS:-"$DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT}"}
```
I don't think `export` is needed on L42.

KateFriedman-NOAA replied:
True, I have removed the `export` @ 7782a02.

aerorahul commented on ush/getdump.sh:
```diff
@@ -5,8 +5,8 @@ export COMPONENT=${COMPONENT:-atmos}
 CDUMP=${2:-""}
-SOURCE_DIR=${3:-$DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}}
-TARGET_DIR=${4:-$ROTDIR/${CDUMP}.${PDY}/$cyc/$COMPONENT}
+SOURCE_DIR=${3:-$DMPDIR/${CDUMP}${DUMP_SUFFIX}.${PDY}/${cyc}/${COMPONENT}}
```
Is export needed on L4?

KateFriedman-NOAA replied:
Nope, have removed @ 7ba270f. Thanks!

aerorahul reviewed (commented):
Some comments for consideration.

KateFriedman-NOAA committed:
Update parm/config/config.anal

Hardcode COMPONENT=atmos in config.anal.

Co-authored-by: Rahul Mahajan <aerorahul@users.noreply.github.com>

KateFriedman-NOAA committed:
Remove unnecessary export for COMPONENT

Removed unnecessary export for the COMPONENT variable in
drive_makeprepbufr.sh

Refs: #802

KateFriedman-NOAA committed:
Remove unneeded export for COMPONENT

Remove unneeded export for COMPONENT in ush_getdump.sh

Refs: #802
